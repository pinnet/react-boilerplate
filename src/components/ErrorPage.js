import React from 'react';
const ErrorPage = () => (
    <div>
    <img src='/images/404.jpg' />
    </div>
);
export default ErrorPage;