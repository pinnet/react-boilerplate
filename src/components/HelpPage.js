import React from 'react';
const HelpPage = () => (
    <div>
    
    </div>
);
export default HelpPage;