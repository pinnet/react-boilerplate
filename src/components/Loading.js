import React from 'react'

const Loading = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader.gif" ></img>
    </div>    
);

export default Loading;