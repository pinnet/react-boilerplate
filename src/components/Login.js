import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from './../actions/auth';

export const Login = ({startLogin}) => (
    <div className="box-layout">
    <div className="box-layout__box">
    <h1 className="box-layout__title"></h1>
    <p></p>
        <button className="button" onClick={startLogin}>Login via Google</button>
    </div>
    </div>
);
const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(Login);