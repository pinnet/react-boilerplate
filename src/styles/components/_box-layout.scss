.box-layout{
    background: url('/images/bg.jpg');
    background-size: cover;
    align-items:center;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    color:$login-text;
    background: fade-out(rgb(85, 146, 224),.25);
    border-radius: 9px;
    text-align: center;
    padding:$xl-size $l-size;

}

.box-layout__title{

    margin:0 0 $l-size 0;
    line-height: 1;
}