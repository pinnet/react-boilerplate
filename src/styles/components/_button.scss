.button{
    background: $mid-blue;
    border:none;
    border-bottom: .2rem solid darken($mid-blue, 10%);
    color: white;
    display:inline-block;
    font-weight: 300;
    font-size: $m-size;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;
}

.button--link{
    background:none;
    border-bottom: none;
}

.button--secondary{
    background: #888;
    border-bottom: .2rem solid darken(#888, 10%);
}