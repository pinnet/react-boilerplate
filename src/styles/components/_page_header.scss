.page-header{
    background:$off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
    @media (max-width: $desktop-breakpoint) {
        font-size: $font-size-small;
    } 
}
.page-header__title{
    margin:0;
    font-weight:300;
    span {
        font-weight: bold;
    }
}
.page_header__actions{
    margin-top: $m-size;
}