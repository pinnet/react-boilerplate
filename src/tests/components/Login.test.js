import React from 'react';
import { Login } from './../../components/Login'
import { shallow } from 'enzyme'; 

it('should render correctly', () => {
    const wrapper = shallow(<Login startLogin={() => {}}/>);
    expect(wrapper).toMatchSnapshot();   
});

it('should login on button click',() => {
    const startLogin = jest.fn();
    const wrapper = shallow(<Login startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toBeCalled();
})