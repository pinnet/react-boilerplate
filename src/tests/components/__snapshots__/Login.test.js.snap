// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    />
    <p />
    <button
      className="button"
      onClick={[Function]}
    >
      Login via Google
    </button>
  </div>
</div>
`;
